supervision/metrics/mean_average_recall.py

```diff
@@ -0,0 +1,452 @@
+            (ObjectSizeCategory.MEDIUM, "medium_objects"),
+            (ObjectSizeCategory.LARGE, "large_objects"),
+        ]:
+            size_predictions = []
+            size_targets = []
+            for predictions, targets in zip(self._predictions_list, self._targets_list):
+                size_predictions.append(
+                    self._filter_detections_by_size(predictions, size_category)
+                )
+                size_targets.append(
+                    self._filter_detections_by_size(targets, size_category)
+                )
+            setattr(result, container, self._compute(size_predictions, size_targets))
+
+        return result
+
+    def _compute(
+        self,
+        predictions_list: List[Detections],
+        targets_list: List[Detections],
+    ) -> MeanAverageRecallResult:
+        if not predictions_list or not targets_list:
+            return MeanAverageRecallResult(
+                metric_target=self._metric_target,
+                is_class_agnostic=self._class_agnostic,
+                mean_average_recall=0.0,
+                ar_per_class=np.array([]),
+                matched_classes=np.array([]),
+            )
+
+        all_recalls = []
+        all_class_ids = []
+        iou_thresholds = np.linspace(0.5, 0.95, 10)
+
+        for predictions, targets in zip(predictions_list, targets_list):
+            if targets.is_empty():
+                continue
+
+            prediction_contents = self._detections_content(predictions)
+            target_contents = self._detections_content(targets)
+
+            if predictions.is_empty():
+                unique_classes = np.unique(targets.class_id)
+                all_class_ids.extend(unique_classes)
+                all_recalls.extend([0.0] * len(unique_classes))
+                continue
+
+            if self._metric_target == MetricTarget.BOXES:
+                iou = box_iou_batch(target_contents, prediction_contents)
+            elif self._metric_target == MetricTarget.MASKS:
+                iou = mask_iou_batch(target_contents, prediction_contents)
+            elif self._metric_target == MetricTarget.ORIENTED_BOUNDING_BOXES:
+                iou = oriented_box_iou_batch(target_contents, prediction_contents)
+            else:
+                raise ValueError("Unsupported metric target for IoU calculation")
+
+            # For each class
+            unique_classes = np.unique(targets.class_id)
+            for class_id in unique_classes:
+                target_mask = targets.class_id == class_id
+                pred_mask = (
+                    predictions.class_id == class_id
+                    if not self._class_agnostic
+                    else slice(None)
+                )
+
+                if not any(target_mask):
+                    continue
+
+                class_iou = iou[target_mask]
+                if any(pred_mask):
+                    class_iou = class_iou[:, pred_mask]
```

Comment:
General comment: I don't see the sorting of predictions by confidence anywhere. I believe we should evaluate the top-1, top-10, top-100 most confident predictions.